File: src/main/java/com/crui/patterns/examples/turnos_medicos/after/CreadorDeDoctores.java
package com.crui.patterns.examples.turnos_medicos.after;

public class CreadorDeDoctores {

  public static Doctor crearCardiologoGeneral(String nombre, String email) {
    return new Doctor(nombre, new Especialidad("Cardiología > General"), email);
  }

  public static Doctor crearNeumonologo(String nombre, String email) {
    return new Doctor(nombre, new Especialidad("Neumonología > General"), email);
  }

  public static Doctor crearAlergista(String nombre, String email) {
    return new Doctor(nombre, new Especialidad("Neumonología > Alergias"), email);
  }

  public static Doctor crearKinesiologo(String nombre, String email) {
    return new Doctor(nombre, new Especialidad("Kinesiología > General"), email);
  }

  public static Doctor crearTraumatologo(String nombre, String email) {
    return new Doctor(nombre, new Especialidad("Kinesiología > Traumatología"), email);
  }

  public static Doctor crearClinicoGeneral(String nombre, String email) {
    return new Doctor(nombre, new Especialidad("Clínica > General"), email);
  }
}

File: src/main/java/com/crui/patterns/examples/turnos_medicos/after/CreadorDeDoctores.py
from Doctor import Doctor
from Especialidad import Especialidad

class CreadorDeDoctores:
    def crearCardiologoGeneral(nombre: str, email: str) -> Doctor:
        return Doctor(nombre, Especialidad("Cardiología > General"), email)

    def crearNeumonologo(self, nombre: str, email: str) -> Doctor:
        return Doctor(nombre, Especialidad("Neumonología > General"), email)

    def crearAlergista(self, nombre: str, email: str) -> Doctor:
        return Doctor(nombre, Especialidad("Neumonología > Alergias"), email)

    def crearKinesiologo(self, nombre: str, email: str) -> Doctor:
        return Doctor(nombre, Especialidad("Kinesiología > General"), email)

    def crearTraumatologo(self, nombre: str, email: str) -> Doctor:
        return Doctor(nombre, Especialidad("Kinesiología > Traumatología"), email)

    def crearClinicoGeneral(self, nombre: str, email: str) -> Doctor:
        return Doctor(nombre, Especialidad("Clínica > General"), email)